bin/cluster_validator.rb

```diff
@@ -0,0 +1,51 @@
+require "cluster"
+require "services"
+Services.mongo!
+
+# Goes through all clusters, checks if they are valid,
+# and prints the first ocn of any invalid cluster to a file.
+
+class ClusterValidator
+  attr_reader :output_path # file name
+  def initialize(buffer_max: 500)
+    @buffer = []
+    @buffer_max = buffer_max
+    # Make an output file in the right place
+    ymd = Time.now.strftime("%Y-%m-%d")
+    dir = Settings.local_report_path
+    FileUtils.mkdir_p(dir)
+    @output_path = "#{dir}/cluster_validator_#{ymd}.txt"
+  end
+
+  def run
+    puts "Writing to #{output_path}"
+    @outf = File.open(output_path, "w")
+    # Go through each cluster and check if valid.
+    comment "These are ocns of invalid clusters:"
+    Cluster.each do |c|
+      unless c.valid?
+        @buffer << c.ocns.first
+      end
+      if @buffer.size > @buffer_max
```

Comment:
Curious what motivated the buffer (i.e. did you try it and find that outputting every time was really slowing things down?) -- which is to say, I would be inclined to remove the buffering unless you have specific evidence it provides a benefit

Reply:
Yeah this is one of those defensive moves arising from the fact that I have no idea how this scales from 10 clusters in dev to the many many clusters in prod. I'm guessing it's not saving us a ton, so it can go away in the name of simplicity.

Reply:
Buffering removed.